html/semantics/forms/the-input-element/input-selection.html

```diff
@@ -0,0 +1,122 @@
+<!DOCTYPE HTML>
+<title>Input element programmatic selection support</title>
+<link rel="author" title="yaycmyk" href="mailto:evan@yaycmyk.com">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/forms.html#dom-textarea/input-select">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<script>
+
+/* all textual, non-hidden inputs support .select() */
+test(function() {
+  var valid = [
+    "text",
+    "search",
+    "url",
+    "tel",
+    "password"
+  ];
```

Comment:
Nope! I know this is confusing, I thought there was a bug in the spec at first, but I think there isn't, it's just confusing.

These are the possibilities:

* `select()` **does not apply** - `InvalidStateError` exception (e.g. `hidden`).
* `select()` **applies**, behavior is one of:
   * Select the text (if the control is a text field, e.g. `text`).
   * Do nothing (otherwise, e.g. `date`).

Look at the "bookkeeping details" for each type to see if it applies or not.